opentracing-mock/src/main/java/io/opentracing/mock/MockTracer.java

```diff
@@ -119,24 +121,29 @@ protected void onSpanFinished(MockSpan mockSpan) {
             public <C> MockSpan.MockContext extract(Format<C> format, C carrier) {
                 Long traceId = null;
                 Long spanId = null;
+                Map<String, String> baggage = new HashMap<>();
 
                 if (carrier instanceof TextMap) {
                     TextMap textMap = (TextMap) carrier;
-                    Iterator<Map.Entry<String, String>> iterator = textMap.iterator();
-                    while (iterator.hasNext()) {
-                        Map.Entry<String, String> entry = iterator.next();
+                    for (Map.Entry<String, String> entry : textMap) {
                         if (TRACE_ID_KEY.equals(entry.getKey())) {
                             traceId = Long.valueOf(entry.getValue());
                         } else if (SPAN_ID_KEY.equals(entry.getKey())) {
                             spanId = Long.valueOf(entry.getValue());
+                        } else {
+                            baggage.put(entry.getKey(), entry.getValue());
                         }
                     }
                 } else {
                     throw new IllegalArgumentException("Unknown carrier");
                 }
 
                 if (traceId != null && spanId != null) {
-                    return new MockSpan.MockContext(traceId, spanId, Collections.<String, String>emptyMap());
+                    if (baggage.isEmpty()) {
```

Comment:
Purely to ensure that the behavior if there are no baggageItems exactly matches current behavior. I didn't want to affect anything that MockContext might do with the empty baggage. It's _probably_ safe to skip the check and just pass the hashmap all the time.